ArkEcosystem/laravel-foundation
# feat: upgrade markdown editor

## Summary

https://app.clickup.com/t/1qa4wyq and https://app.clickup.com/t/1wu12gb

Upgrades the markdown editor scripts to make them compatible with `@toast-ui/editor^3.1.1`, refactor and cleanup

**UPDATE**
This PR also includes dark mode, see note below for more details.

![image](https://user-images.githubusercontent.com/17262776/146095141-e6f72b0e-9bd6-4ecc-b061-71460be15c8f.png)
![image](https://user-images.githubusercontent.com/17262776/146095227-4d6c7961-0a52-4553-8b12-26fb96f01622.png)

![image](https://user-images.githubusercontent.com/17262776/146095360-7491fe60-d53d-4778-9ce6-bdd028e81935.png)
![image](https://user-images.githubusercontent.com/17262776/146095377-a4dd9acf-a545-4c80-9d5c-b9a37673b5e6.png)



**Projects PRs:**
1. MSQ: https://github.com/ArkEcosystem/marketsquare.io/pull/2530
2. ark.io: https://github.com/ArkEcosystem/ark.io/pull/1022



### To test

1. Upgrade the yarn dependency to 3.1.1 `yarn add @toast-ui/editor@3.1.1` on the different projects that use the markdown editor
2. Test the editor, compare design, mobile version, check for errors in the console, etc.
3. Check the editor on all the PRs that I created to update the editor and ensure I am not missing any project

You can test the "full" version of the toolbar in the `ark.io` project:
1. merge foundation, install deps, recompile, etc
2. Go to `kiosk/articles/create`
3. Test
4. notice that the image modal doesn't implement file upload yet, that is in purpose

**Note:** The new design implementation combines [this design](https://www.figma.com/file/z3l29MPdSHU5YGbLeWSJSL/Markdown-Editor?node-id=486%3A2877) and the dark markdown used on the [dark mode design file](https://www.figma.com/file/6VhdbE82NloQmxd6kJiGGN/MSQ?node-id=1063%3A197362), since there are not consistent it may vary in icon size, borders, etc. [Asked Oleg](https://cryptoarkproject.slack.com/archives/CLAGGC7LY/p1639584604471000) and he liked the current implementation but if you have doubts better to ask him.


## Checklist
